src/components/Post/PostDetailComponent.jsx

```diff
@@ -1,16 +1,127 @@
+                  </h1>
+                  <p>
+                    U S E R_S T O K E_R A T I N G: {this.state.post.user_rating}
+                  </p>
+                  <Image src='https://i.pinimg.com/originals/90/64/a1/9064a16dc937d44f1fedba59074d5fa5.jpg' size='large' wrapped />
+                  <h4>
+                    p o s t e d_b y: {this.state.post.user.username}
+                  </h4>
+                  <div className='break-word'>
+                    {this.state.post.post_content}
+                  </div>
+                </Segment>
+                <Segment id='post-detail-map-div'>
+                  <h3>
+                    {`${this.state.post.city}, ${this.state.post.state}`}
+                  </h3>
+                  <p>
+                    l  a  t  : {this.state.post.lat}_
+                      l  n  g  : {this.state.post.lng}
+                  </p>
+                  <div className='google-map-container'>
+                    <GoogleMapReact
+                      bootstrapURLKeys={{ key: process.env.REACT_APP_MAP_API_KEY }}
+                      center={this.state.userLocation}
+                      zoom={this.state.zoom}
+                    >
+                      <MapMarker
+                        lat={this.state.post.lat}
+                        lng={this.state.post.lng}
+                        text={this.state.post.user_rating}
+                        city={this.state.post.city}
+                        color={this.postColorHandler(this.state.post)}
+                        upVote={this.state.post.up_votes}
+                        downVote={this.state.post.down_votes}
+                        date={this.state.post.created_at.substring(0, 10)}
+                      />
+                    </GoogleMapReact>
+                  </div>
+                  <p>
+                    u p d a t e d_A t : {this.state.post.updated_at.substring(0, 10)}
+                  </p>
+                  <div>
+                    u p_V o t e s: {this.state.post.up_votes}
+                  </div>
+                  <div>
+                    d o w n_V o t e s: {this.state.post.down_votes}
+                  </div>
+                  {
+                    localStorage.getItem('user_Id') === this.state.post.user.id
+                      ? (
+                        <Link to={
+                          {
+                            pathname: '/post/' + this.state.post.id + '/edit',
+                            state: {
+                              post: this.state.post,
+                              color: this.postColorHandler(this.state.post)
+                            }
+                          }
+                        }>
+                          <button className="windows-btn-edit">
+                            <span className='window-btn-span-cancel'>
+                              Edit
+                            </span>
+                          </button>
+                        </Link>
```

Comment:
So this is what I've learned at my job and what my client likes to do in terms of code refactoring, but when trying to apply JSX rendering using ternary operators (i.e. `someCondition === value ? ... : ...`), you should not try to rendering too much JSX (but this open for debate). So what my Senior Engineers always told me to do is to place JSX as a separate component method. So for this instance, you can do something like `renderEditButton() {...}` and return the JSX there and then invoke it using the ternary condition:
```
localStorage.getItem('user_Id') === this.state.post.user.id ? this.renderEditButton() : null
```

The reason for this is because...
1. You NEVER know when you will be reusing that `Edit` button JSX component and you might use it elsewhere.
2. It prevents the JSX code from being too cluttered.

Reply:
yea 100% agree with this in terms of scalability and readability,
but is it open to debate because of efficiency/performance?

Reply:
Performance is dictated on how much memory is being allocated, the types of data structures being used and the algorithms being implemented.

But React JSX at the end of the day gets compiled to Javascript. This case above is more for code readability and the potential prevention of writing repeatable code.